### DISCUSSION TITLE
Clcikhouse replica selection mechanism   |    Replica expansion

### DISCUSSION THREAD

**Discussion by honobi** [ClickHouse/ClickHouse#66316] [Q&A] [answered]
Situation description:
I have a single-shard, single-replica ClickHouse cluster using the Distributed+ReplicatedMergeTree table engine.
I added a new node to the cluster, and the config file of the new node contains the config information of all three replicas, while the config files of the two old nodes only contain the config information of the two old nodes.

My question:
When executing a SELECT query on replica 1, if the data in the newly added replica is not up-to-date, or significantly behind the old replicas, will the newly added replica be selected?

**Comment by yiyang-shao** [comment, marked as answer]
Hi there,

Could you check if this [max_replica_delay_for_distributed_queries setting](https://clickhouse.com/docs/en/operations/settings/settings#max_replica_delay_for_distributed_queries) is relevant to your use case?
